src/component/handlers/edit/commands/keyCommandBackspaceToStartOfLine.js

```diff
@@ -24,18 +25,24 @@ function keyCommandBackspaceToStartOfLine(
   var afterRemoval = removeTextWithStrategy(
     editorState,
     strategyState => {
+      var selection;
       var domSelection = global.getSelection();
       var range = domSelection.getRangeAt(0);
-      range = expandRangeToStartOfLine(range);
-
-      var selection = getDraftEditorSelectionWithNodes(
-        strategyState,
-        null,
-        range.endContainer,
-        range.endOffset,
-        range.startContainer,
-        range.startOffset
-      ).selectionState;
+
+      if (range.collapsed && range.startOffset === 0) {
```

Comment:
Instead of obtaining this from the DOM selection, you could look it up from `strategyState`.

```js
const selectionState = strategyState.getSelection();
if (selectionState.isCollapsed() && selectionState.getStartOffset() === 0) {
  return moveSelectionBackward(strategyState, 1);
}

// do what it was doing before
const domSelection = ...;
const range = ...;
// etc.
```

This way, you can avoid even looking into the DOM selection for the case in question.

Also, you should operate with `strategyState` (which is an `EditorState`) rather than the outer `editorState`. It should be the same object, but in theory it could be different.